localstack/services/stepfunctions/stepfunctions_starter.py

```diff
@@ -107,9 +108,20 @@ def check_stepfunctions(expect_shutdown=False, print_error=False):
         assert out and isinstance(out.get("stateMachines"), list)
 
 
-def restart_stepfunctions():
-    if not PROCESS_THREAD:
+def restart_stepfunctions(persistence_path: Optional[str] = None):
+    if not PROCESS_THREAD or not PROCESS_THREAD.process:
         return
     LOG.debug("Restarting StepFunctions process ...")
+    pid = PROCESS_THREAD.process.pid
     PROCESS_THREAD.stop()
-    start_stepfunctions(asynchronous=True)
+    _wait_for_process_to_be_killed(pid)
+    start_stepfunctions(persistence_path=persistence_path)
+
+
+def _wait_for_process_to_be_killed(pid):
```

Comment:
nit: This looks like a useful utility we could put into `localstack/utils/run.py`. 👍 (exposing some public API like `wait_for_process_killed(pid: int, sleep: float=None, retries: int=None)`)